internal/controller/pkg/resolver/reconciler.go

```diff
@@ -66,13 +68,25 @@ const (
 	errFmtMissingDependency = "missing package (%s) is not a dependency"
 	errInvalidConstraint    = "version constraint on dependency is invalid"
 	errInvalidDependency    = "dependency package is not valid"
-	errFindDependency       = "cannot find dependency version"
+	errFindDependency       = "cannot find dependency version to install"
 	errGetPullConfig        = "cannot get image pull secret from config"
 	errFetchTags            = "cannot fetch dependency package tags"
-	errNoValidVersion       = "cannot find a valid version for package constraints"
-	errFmtNoValidVersion    = "dependency (%s) does not have version in constraints (%s)"
+	errFindDependencyUpdate = "cannot find dependency version to update"
+	errFmtNoValidVersion    = "dependency (%s) does not have a valid version to upgrade that satisfies all constraints. If there is a valid version that requires downgrade, manual intervention is required. Constraints: %v"
 	errInvalidPackageType   = "cannot create invalid package dependency type"
+	errGetDependency        = "cannot get dependency package"
 	errCreateDependency     = "cannot create dependency package"
+	errUpdateDependency     = "cannot update dependency package"
+)
+
+// Event reasons.
+const (
+	reasonErrBuildDAG       event.Reason = "BuildDAGError"
+	reasonCyclicDependency  event.Reason = "CyclicDependency"
+	reasonInvalidDependency event.Reason = "InvalidDependency"
+	reasonNoValidVersion    event.Reason = "NoValidVersion"
+	reasonErrCreate         event.Reason = "CreateError"
+	reasonErrUpdate         event.Reason = "UpdateError"
```

Comment:
Could be a follow up, but, I am starting to think to have a condition on the lock objects instead or in addition 🤔